erroranalysis/erroranalysis/analyzer/error_analyzer.py

```diff
@@ -445,6 +451,10 @@ class and for the regression case a method of predict()
     :type feature_names: numpy.array or list[]
     :param categorical_features: The categorical feature names.
     :type categorical_features: list[str]
+    :param metadata_columns: The set of columns that are not passed
+        to the model or explainers. These columns can be used for
+        other analyses.
+    :type metadata_columns: list[str]
```

Comment:
optional[list[str]] if `None` is valid?

Reply:
Although we haven't done that on the others

Reply:
I'm actually taking the unrelated doc adjustments out of this PR into a separate one. Perhaps I can do that there!

Reply:
#1214 !

Also, the optional[list[str]] annotations are only used for type annotations on the args in code, not in the docstring, right? I may be wrong...

Reply:
added the annotations, but there will probably be a few merge conflicts to resolve once #1214  is merged.